hw/xbox/xid-sdl.c

```diff
@@ -511,18 +486,49 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
     SDL_JoystickEventState(SDL_ENABLE);
 #endif
 
-#ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
-    if (s->sdl_haptic == NULL) {
-        fprintf(stderr, "Joystick doesn't support haptic\n");
-    } else {
-        if ((SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) == 0) {
-            fprintf(stderr, "Joystick doesn't support necessary haptic effect\n");
-            SDL_HapticClose(s->sdl_haptic);
-            s->sdl_haptic = NULL;
+    if (!SDL_HapticOpened(s->device_index)) {
+        s->sdl_haptic = SDL_HapticOpen(s->device_index);
+        if (s->sdl_haptic == NULL) {
+            fprintf(stderr, "Joystick doesn't support haptic effects\n");
+        } else {
+            if (SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) {
```

Comment:
flipped the `if` statement to allow adding `else if` later for other fallback haptic modes in future commit.

Reply:
Noted. Not worth mentioning as the intention is clear.

*Generally, unless there's an issue, don't bother adding a GitHub comment (to avoid unnecessary discussions or many "pings" to maintainers). If something is unclear, the reviewers will leave a comment anyway. You can then state the intention (which is often asked to be moved into code comments, if it's hard to understand).*